lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -160,6 +160,7 @@ def warn_about_bare_error
                         "will match when Ruby raises a `NoMethodError`, `NameError` or " \
                         "`ArgumentError`, potentially allowing the expectation to pass " \
                         "without even executing the method you are intending to call. " \
+                        "Actual error raised was #{description_of(@actual_error)}. "\
```

Comment:
@xaviershay initially it does work but when you implement your solution and make `expect(message).not_to match ex` pass, the second test fails. The reason why it fails is because it expects an error to be raised;
```
1) expect { ... }.to raise_error issues a warning that does not include current error when its not present
     Failure/Error: RSpec::Support.notify_failure(RSpec::Expectations::ExpectationNotMetError.new message)
       expected Exception but nothing was raised
   ```